Products/ExternalEditor/ExternalEditor.py

```diff
@@ -177,7 +177,7 @@ def index_html(self, REQUEST, RESPONSE, path=None):
             # We've been requested to send only the metadata. The
             # client will presumably fetch the data itself.
             self._write_metadata(RESPONSE, metadata, metadata_len)
-            return ''
+            return metadata + '\n'
```

Comment:
This change looks unrelated to the PR.  Given that the previous line has already written the metadata to the response stream, I'm not sure it is correct, either.

Reply:
It was necessary to get the metadata into the response. The last line of the method does it the same way: [return join((metadata, body), '\n')](https://github.com/zopefoundation/Products.ExternalEditor/blob/master/Products/ExternalEditor/ExternalEditor.py#L240) (but `body` is empty here.)

I am not sure why this ever can have worked. I should investigate what [RESPONSE.write(metadata)](https://github.com/zopefoundation/Products.ExternalEditor/blob/master/Products/ExternalEditor/ExternalEditor.py#L270) does and why it is not enough; maybe even in contrast to Zope 2.13 on the 1.1.x branch.

Reply:
@tseaver You were right: There is a bug in `WSGIPublisher.py` where the wrong `stdout` stream is read from if using a predefined response object as `Testing.ZopeTestCase.zopedoctest.functional.http` does. See zopefoundation/Zope#256.